homeassistant/core.py

```diff
@@ -747,6 +764,8 @@ async def async_stop(self, exit_code: int = 0, *, force: bool = False) -> None:
         self.state = CoreState.not_running
         self.bus.async_fire(EVENT_HOMEASSISTANT_CLOSE)
 
+        for task in running_tasks:
```

Comment:
if we cancel them, should we then consider awaiting them ? I also wonder if this then should happen as part of stage 2 where we close ?

Reply:
Would make sense to await them guarding for canceled error with a broad exception catch that logs

Reply:
f0dae019452c45e4c7a44626544517d292df3b38
0e31f8c93edbd51bad6b46180ea9c38ac12037f9